tests/functional/karate_tests/src/test/java/com/karate/openwhisk/smoketests/SmokeTestRunner.java

```diff
@@ -0,0 +1,57 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.karate.openwhisk.smoketests;
+import static org.junit.Assert.assertTrue;
+
+import java.io.File;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+
+import org.apache.commons.io.FileUtils;
+import org.junit.Test;
+
+import com.intuit.karate.cucumber.CucumberRunner;
+import com.intuit.karate.cucumber.KarateStats;
+
+import cucumber.api.CucumberOptions;
+import net.masterthought.cucumber.Configuration;
+import net.masterthought.cucumber.ReportBuilder;
+
+@CucumberOptions(tags = {"~@ignore","~@driver","~@reliability","~@resiliency","~@concurrent","~@wskfunctions"})
+public class SmokeTestRunner {
+    @Test
+    public void testwskFunctions() {
+
+        String karateOutputPath = "target/surefire-reports";
+        KarateStats stats = CucumberRunner.parallel(getClass(), 1, karateOutputPath);
+        generateReport(karateOutputPath);
+        assertTrue("there are scenario failures", stats.getFailCount() == 0);
+    }
+    private static void generateReport(String karateOutputPath) {
```

Comment:
generateReport logic looks common in SmokeTestRunner and other runner class. Move them to a base class and reuse

Reply:
Resolved and created Base class in utils package.